tutorials/qubit-dynamics.md

```diff
@@ -0,0 +1,192 @@
+
+### Introduction
+In this notebook we will explore the dynamics of a single-qubit interacting with an environment. The evolution of the qubit state is governed by the Master equation. We will make use of the master equation solver `qutip.mesolve` implemented in qutip, to obtain the time-evolution of the qubit for different settings.
+
+You can read more about the master equation solver (and the theory behind it) in the [QuTiP docs](https://qutip.org/docs/latest/apidoc/functions.html?highlight=sesolve#module-qutip.sesolve).
+
+### Import
+Here we import the required modules for this example.
+
+```python
+%matplotlib inline
+import matplotlib.pyplot as plt
+import numpy as np
+from qutip import sigmax, sigmay, sigmaz, mesolve, basis, sigmam, Bloch
+```
+
+### System setup
+We will start with a basic Hamiltonian for the qubit, which flips the state of the qubit represented by the Pauli Matrix $\sigma_x$.
+
+$H = \frac{\Delta}{2} \sigma_x$
+
+Additionally, we add a collapse operator that describes the dissipation of energy from the qubit to an external environment. The collapse operator is defined by
+
+$C = \sqrt{g} \sigma_z$
+
+where $g$ is the dissipation coefficient.
+We define the qubit to be in the ground state at $t=0$.
+
+```python
+# coefficients
+delta = 2 * np.pi
+g = 0.25
+
+# hamiltonian
+H = delta / 2.0 * sigmax()
+
+# list of collapse collapse operator
+c_ops = [np.sqrt(g)*sigmaz()]
+
+# initial state
+psi0 = basis(2,0)
+
+# times 
+tlist = np.linspace(0,5,100)
+```
+
+### Time-evolution
+We pass these definition to the `qutip.mesolve` function. The collapse operators need to be passed in a list (even if there is only one collapse operator). As the fifth argument we pass a list of operators, for which the solver will return the expectation value at the given times in `tlist`. In this example we want to obtain the expectation value for $\sigma_z$.
+
+```python
+res = mesolve(H, psi0, tlist, c_ops, [sigmaz()])
+```
+
+For this particular Hamiltonian and dissipation process we can derive the analytical solution for the expectation value of $\sigma_z$.
+
+```python
+sz_analytic = np.cos(2*np.pi*tlist) * np.exp(-tlist * g)
+```
+
+### Comparison to Analytical Solution
+By plotting the expectation value from `mesolve` and the analytical result we can verify the correctness of the result. We can access the expectation value generated by `mesolve` by accessing `result.expect[0]`.
+
+```python
+plt.scatter(tlist, res.expect[0], c='r', marker='x', label='mesolve')
+plt.plot(tlist, sz_analytic, label='Analytic')
+plt.xlabel('Time'), plt.ylabel('<sigma_z>')
+plt.legend();
+```
+
+## Qubit Dynamics on the Bloch Sphere
+
+We can also visualise the dynamics of the qubit state on the Bloch sphere. To generate more interesting plots, we consider slightly more complex dynamics of the qubit state.
+
+### Rotating State
+
+Consider the following Hamiltonian: 
+
+$H = \Delta ( \, cos(\theta) \, \sigma_z + sin(\theta) \, \sigma_x  \, )$.
```

Comment:
in the markdown version these latex spacing commands \, get a bit mangled. seems ok in the notebook version.